rse-project.Rmd

```diff
@@ -21,7 +21,214 @@ small pieces in predictable places with readable names are easier to find and us
+    This file usually contains a plain text citation,
+    and may also include entries formatted for various bibliographic systems like [BibTeX][bibtex].
+    Some projects also have a separate `CONTRIBUTORS` file listing everyone who has contributed to it,
+    while others include this as a section in `CITATION`.
+
+If setting up the project or contributing to it are complex,
+there may also be a file called `BUILD` that explains how to install the required software,
+the formatting conventions for new data entries,
+and so on.
+These instructions can also be included as a section in `README`.
+Whichever convention is used,
+remember that the easier it is for people to get set up and contribute,
+the more likely they are to do so @Stei2014.
+
+TODO: `README` or `CONTRIBUTING` are also great places to document the conventions of structure (the rest of this chapter), naming and style.
+
+## How do I document a project?
+
+TODO: What does someone need to know beyond the collection of files in the project? 
+ 
+TODO: What should go in a `README`? Use Elizabeth's Step 4: "Administrative and personnel details" and "Project information" prompts as a guide. https://www.ideals.illinois.edu/bitstream/handle/2142/91611/W2%20-%20DataDocumentationHandout.docx
+
+## How should I structure the files and directories in my project? 
+
+TODO: "There are many strongly held opinions for how this should be done. These opinions are strongly held not becasue the authors think their way is **the** right way, but because its more important to be consistent, and it's easier to be consistent when you have publicly decreed that *this is* the way you will do something.  Does it matter which you use? Not really.  More important to use a structure that works for you, your audience and your project, be consistent, and communicate your conventions."
+
+TODO: "Here are three examples of opinionated project structure." Design exercise that encourages comparing these structures, what do they have in common? 
+
+[Packaging data analytical work reproducibly using R (and friends)](https://peerj.com/preprints/3192/)
+
+> "1. Organizes files using conventions in the field
+> 2. Keep data, method and outputs separate while maintaining the relationship between them."
+
+```
+├── DESCRIPTION
+├── README.md
+├── LICENSE
+├── data
+│   └── my_data.csv
+└── analysis
+    └── my_report.Rmd
+```
+
+[Good enough practices in scientific computing](https://doi.org/10.1371/journal.pcbi.1005510)
+
+```
+├── CITATION
+├── README
+├── LICENSE 
+├── requirements.txt
+├── data
+│   └──birds_count_table.csv 
+├── doc
+|   ├── notebook.md
+|   ├── manuscript.md
+|   └── changelog.txt 
+├── results
+|   └── summarized_results.csv 
+└── src
+    ├── sightings_analysis.py 
+    └── runall.py
+```
+
+https://drivendata.github.io/cookiecutter-data-science/
+
+> "Well organized code tends to be self-documenting in that the organization itself provides context for your code without much overhead."
```

Comment:
I personally think this structure below is way too complicated...

Reply:
But I'm also not a Python person, so :man_shrugging:

Reply:
I also agree!  And I included for exactly that reason - it's a a good example of "a strongly held opinion" that may not be right for you.  I.e. most projects I work on (and I suspect many others) don't need anywhere near this kind of complexity in structure, but it obviously works for them and their projects.